brave/brave-ios
# Fix #7352: Add tracking and ad block level setting

## Summary of Changes


This pull request fixes #7352 

## Submitter Checklist:


## Test Plan:

1. Test any known cosmetic filtering for pages like reddit, Some known test are:
2. Go to brave search and search for "trezor". 
- The ad on the top of the page should be hidden on aggressive
- The ad on the top of the page should be shown on standard
3. Enable "Block cookie consent notices" and go to reddit. 
- Cookie notices should be blocked on standard and aggressive
5. Enable "Block open in app notices" and go to reddit.
- Open in app notices should be blocked on standard and aggressive
7. Go to the following page: https://dev-pages.brave.software/filtering/cosmetic-filtering.html
- On aggressive mode, all images should be hidden after clicking "Run tests"
- On aggressive mode, only 3rd party images should be hidden after clicking "Run tests" (there is an unhiding delay)

## Screenshots:

![Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 59](https://github.com/brave/brave-ios/assets/909331/41721dcd-63b2-4135-832d-d385c856b315)
![Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 53](https://github.com/brave/brave-ios/assets/909331/f388a4fb-1da2-48be-932c-d666809d3d79)
![Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 43](https://github.com/brave/brave-ios/assets/909331/cf34cc77-c8d6-4fc5-8a8d-2d459759bc37)
![Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 36](https://github.com/brave/brave-ios/assets/909331/38f95ae7-a740-4c14-95b0-d5fda7a37036)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
